netkan/netkan/metadata.py

```diff
@@ -121,3 +122,164 @@ def licenses(self):
             return lic
         else:
             return [lic]
+
+    class Version:
+
+        Pattern = re.compile("^(?:(?P<epoch>[0-9]+):)?(?P<version>.*)$")
+
+        def __init__(self, version_string):
+            self.string = version_string
+            match = self.Pattern.fullmatch(self.string)
+            if not match:
+                raise Exception('Invalid version format')
+            if match.group('epoch') and match.group('epoch').isnumeric():
+                self.epoch = int(match.group('epoch'))
+            else:
+                # https://github.com/KSP-CKAN/CKAN/blob/master/Spec.md#epoch
+                # 0 assumed if no epoch supplied
+                self.epoch = 0
+            self.bare_version = match.group('version')
+            if self.bare_version is None:
+                raise Exception
+
+        def __eq__(self, other):
+            return not self < other and not self > other
```

Comment:
Could we compare `self.string == other.string` here? I think that would always give the right answer.

Reply:
This would break `test_issue1076()`

Reply:
Ahh, I see.